vault/mount.go

```diff
@@ -346,6 +346,8 @@ type MountEntry struct {
 	Version        string `json:"plugin_version,omitempty"`         // The semantic version of the mounted plugin, e.g. v1.2.3.
 	RunningVersion string `json:"running_plugin_version,omitempty"` // The semantic version of the mounted plugin as reported by the plugin.
 	RunningSha256  string `json:"running_sha256,omitempty"`
+
+	IsExternalPlugin bool `json:"is_external_plugin,omitempty"` // Whether the mounted plugin is external
```

Comment:
Looking at this, I can't remember why we needed the `logical.Externaler` interface, as I'm pretty sure if the `RunningSha256` is non-empty, that means it's external, and otherwise it's builtin 🤔 Sorry this isn't terribly actionable at the moment, I'll need to dig in a bit further.

Reply:
It might have been useful in an earlier iteration of the code to be able to query the backend for it is external, rather than relying on the mount table. But, I think you're right -- it looks like all uses of `logical.Externaler` are after we have the `Sha256` calculated, so we could refactor `logical.Externaler` out. (Probably not in this PR.)

Reply:
Thanks for confirming. In that case, I'd probably suggest not adding a field to this struct, but instead just add a helper method to compute it:

```go
func (me *MountEntry) IsExternalPlugin() bool {
	return me.RunningSha256 != ""
}
```

Reply:
Thanks!